woocommerce/woocommerce-ios
# [Tooling] Fix GitHub releases for intermediate betas

### Why?

_Similar PR than WPiOS's https://github.com/wordpress-mobile/WordPress-iOS/pull/16674_

Fix the GitHub releases drafts being incorrect for intermediate betas:
 - They used the final version number (`x.y`) for the release name and tag instead of the build number (`x.y.0.N`)
 - They had the incorrect release notes (used the previous ones for any beta, including the first one, created before we manually update the notes with Editorial copy)
 - They didn't have the "Pre-release" checkbox checked on them


### To Test

Not easy to test from start to finish without creating a new beta… and we already did the `finalize_release` for this Sprint so we should not create a new beta even just for testing, so not sure if there's a way to dry-run/test this 😒 

Best way I can think of is comment all the steps in the `Fastfile` that have side effects on `build_and_upload_itc` (namely `gym`, `testflight` and `sentry`) and trigger the lane with the same parameters that are used in the `.circleci/config.yml` for beta build, and check the result. But that will only test a tiny portion of the changes, like it won't test that the release notes extraction works to use the right release notes for first betas nor that the round trip from "run lane on your Mac -> trigger CI -> CI run its own lane" doesn't break mid-chain…

So maybe the best way to test this besides just reviewing the code is to test this during the next cycle…
